server/athenian/api/models/web/goal_template.py

```diff
@@ -0,0 +1,70 @@
+from athenian.api.models.web.base_model_ import Model
+from athenian.api.models.web.jira_metric_id import JIRAMetricID
+from athenian.api.models.web.pull_request_metric_id import PullRequestMetricID
+from athenian.api.models.web.release_metric_id import ReleaseMetricID
+
+GoalTemplateMetricID = JIRAMetricID | PullRequestMetricID | ReleaseMetricID
+
+
+class GoalTemplate(Model):
+    """A template to generate a goal."""
+
+    attribute_types = {
+        "id": int,
+        "name": str,
+        "metric": GoalTemplateMetricID,
+    }
+
+    def __init__(self, id: int = None, name: str = None, metric: GoalTemplateMetricID = None):
+        """GoalTemplate - a model defined in OpenAPI
+
+        :param id: The id of this GoalTemplate.
+        :param name: The name of this GoalTemplate.
+        :param metric: The metric of this GoalTemplate.
+        """
+        self._id = id
+        self._name = name
+        self._metric = metric
+
+    @property
+    def id(self) -> int:
+        """Gets the id of this GoalTemplate."""
+        return self._id
+
+    @id.setter
+    def id(self, id: int) -> None:
+        """Sets the id of this GoalTemplate."""
+        if id is None:
+            raise ValueError("Invalid value for `id`, must not be `None`")
+
+        self._id = id
+
+    @property
+    def name(self) -> str:
+        """Gets the name of this GoalTemplate."""
+        return self._name
+
+    @name.setter
+    def name(self, name: str) -> None:
+        """Sets the name of this GoalTemplate."""
+        if name is None:
+            raise ValueError("Invalid value for `name`, must not be `None`")
+        if name is not None and len(name) < 1:
+            raise ValueError(
+                "Invalid value for `name`, length must be greater than or equal to `1`",
+            )
+
+        self._name = name
+
+    @property
+    def metric(self) -> GoalTemplateMetricID:
+        """Gets the metric of this GoalTemplate."""
+        return self._metric
+
+    @metric.setter
+    def metric(self, metric: GoalTemplateMetricID) -> None:
+        """Sets the metric of this GoalTemplate."""
+        if metric is None:
+            raise ValueError("Invalid value for `metric`, must not be `None`")
+
+        self._metric = metric
```

Comment:
We must check if the value is in `JIRAMetricID | PullRequestMetricID | ReleaseMetricID`

Reply:
Added.
I just noticed that typing with `base_model_.Enum` is not working as expected, i.e. `metric` is really string.
This is another global issue we must resolve before introducing mypy

Reply:
The string goes directly from aiohttp/connexion, the JSONSchema engine checks enums but doesn't do anything with types, I think. On the other hand, the code is simple now and it's important.